core/src/main/java/org/elasticsearch/common/io/stream/StreamInput.java

```diff
@@ -571,4 +600,10 @@ public static StreamInput wrap(byte[] bytes, int offset, int length) {
         return new InputStreamStreamInput(new ByteArrayInputStream(bytes, offset, length));
     }
 
+    /**
+     * Wraps a {@link StreamInput} and enhances it so that it knows how to deserialize {@link NamedWriteable} arbitrary objects.
+     */
+    public static StreamInput wrap(StreamInput streamInput, NamedWriteableRegistry namedWriteableRegistry) {
```

Comment:
I would put it on the namedwriteablestreaminput rather than here? Or maybe we don't actually need this method, this would just be the constructor?